hack/verify-generated-defaulters.sh broken on base_ami

Seen in many of the latest runs like [here](https://ci.openshift.redhat.com/jenkins/view/AMIs/job/base_ami/2659/consoleFull):

```
===== Verifying Generated Defaulters =====
FAILURE: Verification of defaulters failed:
[WARNING] No compiled `gendefaults` binary was found. Attempting to build one using:
[WARNING]   $ hack/build-go.sh tools/gendefaults
++ Building go targets for linux/amd64: tools/gendefaults
++ Placing binaries
/data/src/github.com/openshift/origin/hack/build-go.sh took 3 seconds
[ERROR] PID 23313: hack/update-generated-defaulters.sh:17: `INPUT_DIRS=$(
[ERROR]   find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | 	xargs grep --color=never -l '+k8s:defaulter-gen=' | 	xargs -n1 dirname | 	sed "s,^\.,${PREFIX}," | 	sort -u | 	paste -sd,
[ERROR] )` exited with status 123.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/update-generated-defaulters.sh:17: `INPUT_DIRS=$(
[INFO]   find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | 	xargs grep --color=never -l '+k8s:defaulter-gen=' | 	xargs -n1 dirname | 	sed "s,^\.,${PREFIX}," | 	sort -u | 	paste -sd,
[INFO] )`
[INFO]   Exiting with code 123.
```

@ncdc pass this on to someone to own please

--- a/hack/update-generated-defaulters.sh
+++ b/hack/update-generated-defaulters.sh
@@ -5,15 +5,14 @@ os::build::setup_env
 
 os::util::ensure::built_binary_exists 'gendefaults'
 
-PREFIX=github.com/openshift/origin
+PREFIX="${OS_GO_PACKAGE}/"
 
 INPUT_DIRS=$(
-  find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | \
-	xargs grep --color=never -l '+k8s:defaulter-gen=' | \
-	xargs -n1 dirname | \
-	sed "s,^\.,${PREFIX}," | \
-	sort -u | \
-	paste -sd,
+  grep --color=never -rl '+k8s:defaulter-gen=' pkg | \
+  xargs -n1 dirname | \
+  sed "s,^,${PREFIX}," | \
+  sort -u | \
+  paste -sd,
 )
 
 EXTRA_PEER_DIRS=$(
@@ -25,8 +24,8 @@ EXTRA_PEER_DIRS=$(
 )
 
 gendefaults \
-	--logtostderr \
-	--output-base="${GOPATH}/src" \
-	--input-dirs ${INPUT_DIRS[@]} \
-  --extra-peer-dirs ${EXTRA_PEER_DIRS[@]} \
-	"$@"
+  --logtostderr \
+  --output-base="${GOPATH}/src" \
+  --input-dirs "${INPUT_DIRS}" \
+  --extra-peer-dirs "${EXTRA_PEER_DIRS}" \
+  "$@"

--- a/hack/update-generated-listers.sh
+++ b/hack/update-generated-listers.sh
@@ -5,33 +5,18 @@ os::build::setup_env
 
 os::util::ensure::built_binary_exists 'genlisters'
 
-PREFIX=github.com/openshift/origin
+PREFIX="${OS_GO_PACKAGE}/"
 
 INPUT_DIRS=$(
-  find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | \
-	xargs grep --color=never -l '+k8s:defaulter-gen=' | \
-	xargs -n1 dirname | \
-	sed "s,^\.,${PREFIX}," | \
-	sort -u | \
-	paste -sd,
+  grep -rl genclient=true pkg | \
+  xargs -n1 dirname | \
+  grep -v 'pkg\/security\/api' | \
+  sort -u | \
+  sed "s,^,${PREFIX}," | \
+  paste -sd,
 )
-
-INPUT_DIRS=(
-  $(
-    cd ${OS_ROOT}
-    find pkg -name \*.go | \
-      xargs grep -l genclient=true | \
-      xargs -n1 dirname | \
-      sort -u | \
-      grep -v 'pkg\/security\/api'
-  )
-)
-
-INPUT_DIRS=(${INPUT_DIRS[@]/#/github.com/openshift/origin/})
-INPUT_DIRS=$(IFS=,; echo "${INPUT_DIRS[*]}")
-
 
 genlisters \
-	--logtostderr \
-	--input-dirs ${INPUT_DIRS[@]} \
-	"$@"
+  --logtostderr \
+  --input-dirs "${INPUT_DIRS}" \
+  "$@"

--- a/hack/update-generated-openapi.sh
+++ b/hack/update-generated-openapi.sh
@@ -5,24 +5,22 @@ os::build::setup_env
 
 os::util::ensure::built_binary_exists 'genopenapi'
 
-ORIGIN_PREFIX=github.com/openshift/origin
+ORIGIN_PREFIX="${OS_GO_PACKAGE}/"
 
 INPUT_DIRS=(
   # kube apis
   $(
-    find ./vendor/k8s.io/kubernetes -name \*.go | \
-    xargs grep --color=never -l '+k8s:openapi-gen=' | \
+    grep --color=never -rl '+k8s:openapi-gen=' vendor/k8s.io/kubernetes | \
     xargs -n1 dirname | \
-    sed "s,^\./vendor/,," | \
+    sed "s,^vendor/,," | \
     sort -u
   )
 
   # origin apis
   $(
-    find . \( -path ./vendor -o -path ./_output \) -prune -type f -o -name \*.go | \
-    xargs grep --color=never -l '+k8s:openapi-gen=' | \
+    grep --color=never -rl '+k8s:openapi-gen=' pkg | \
     xargs -n1 dirname | \
-    sed "s,^\.,${ORIGIN_PREFIX}," | \
+    sed "s,^,${ORIGIN_PREFIX}," | \
     sort -u
   )
 )
@@ -30,8 +28,8 @@ INPUT_DIRS=(
 INPUT_DIRS=$(IFS=,; echo "${INPUT_DIRS[*]}")
 
 genopenapi \
-	--logtostderr \
-	--output-base="${GOPATH}/src" \
-	--input-dirs "${INPUT_DIRS}" \
-	--output-package "${ORIGIN_PREFIX}/pkg/openapi" \
-	"$@"
+  --logtostderr \
+  --output-base="${GOPATH}/src" \
+  --input-dirs "${INPUT_DIRS}" \
+  --output-package "${ORIGIN_PREFIX}pkg/openapi" \
+  "$@"

